java/dagger/hilt/android/plugin/src/main/kotlin/dagger/hilt/android/plugin/AndroidEntryPointClassVisitor.kt

```diff
@@ -207,10 +213,10 @@ class AndroidEntryPointClassVisitor(
         ClassReader.SKIP_CODE or ClassReader.SKIP_DEBUG or ClassReader.SKIP_FRAMES
       )
       return@use hasMarker
-    }
+    } ?: false
 
-  private fun findAdditionalClassFile(className: String) =
-    File(additionalClasses, "$className.class")
+  private fun findAdditionalClassFile(className: String): File? =
+    additionalClasses.takeIf { it.count() == 1 }?.singleFile?.let { File(it, "$className.class") }
```

Comment:
Since the use of `FileCollection` is a workaround but when actually looking into the directory we expect a single directory (file) in the collection, should we then just use `.singleFile()` without the `takeIf` so that it errors out if there is zero or more than one file. It would also make this method return not-null File

Reply:
After a conversation in the Gradle community slack, I've actually come to reconsider this fix. 

`@Input`, `@InputFiles` etc. are used to determine if the task is up-to-date and if there is a cache miss. Since the visitor is part to the `TransformClassesWithAsmTask` inputs, and that class already defines the full classpath as an input property, this additional property can actually be ignored as part of the cache key by marking it as `@Internal` instead of `@File`. That way there are zero behavior changes in this visitor itself, but the relocatability test still succeeds.

Reply:
Awesome, thanks for the update.